### ISSUE TITLE
Drag and drop issue

### ISSUE THREAD

**Issue by gauravgandhi15** [appium/appium#4695]
I am working on a app where i need to drag and drag the elements to the top of the list. And i am using TouchAction class to perform this functions. This is what i have tried so far:

    action.press(list.get(i)).waitAction(3000).moveTo(16,146).release();
    action.perform();

    action.tap(list.get(i)).waitAction(3000).moveTo(16,146).release();
    action.perform();

    action.longPress(list.get(i)).moveTo(16,146).release();
    action.perform();
 
Nothing worked for me. For first two cases i am getting this error. Look at the logs:

fo: [debug] Pushing command to appium work queue: ["element:click",{"elementId":"8"}]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:click","params":{"elementId":"8"}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: click
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
info: [debug] Pushing command to appium work queue: ["element:touchMove",{"x":16,"y":146}]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:touchMove","params":{"x":16,"y":146}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: touchMove
info: [debug] [BOOTSTRAP] [debug] Display bounds: [0,0][720,1184]
info: [debug] [BOOTSTRAP] [debug] Performing TouchMove using element? false x: 16, y: 146
info: [debug] [BOOTSTRAP] [debug] Updating class "class com.android.uiautomator.core.UiDevice" to enable field "mUiAutomationBridge"
info: [debug] [BOOTSTRAP] [debug] Updating class "class com.android.uiautomator.core.UiAutomatorBridge" to enable field "mInteractionController"
info: [debug] [BOOTSTRAP] [debug] Finding methods on class: class com.android.uiautomator.core.InteractionController
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"Failed to execute touch event","status":13}
info: [debug] Responding to client with error: {"status":13,"value":{"message":"An unknown server-side error occurred while processing the command.","origValue":"Failed to execute touch event"},"sessionId":"6976a8db-cf07-4e3b-99ed-75413ea1cb47"}
info: <-- POST /wd/hub/session/6976a8db-cf07-4e3b-99ed-75413ea1cb47/touch/perform 500 3180.372 ms - 198 
info: --> DELETE /wd/hub/session/6976a8db-cf07-4e3b-99ed-75413ea1cb47 {}
info: Shutting down appium session
info: [debug] Pressing the HOME button
info: [debug] executing cmd: /Users/kumarg/Downloads/android-sdk-macosx/platform-tools/adb -s T065000751 shell "input keyevent 3"
info: [debug] Stopping logcat capture
info: [debug] Logcat terminated with code null, signal SIGTERM
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"shutdown"}
info: [debug] [BOOTSTRAP] [debug] Got command of type SHUTDOWN
info: [debug] Sent shutdown command, waiting for UiAutomator to stop...
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"OK, shutting down","status":0}
info: [debug] [BOOTSTRAP] [debug] Closed client connection

And, for the last one it just move to the next test case without actually performing any issue.

**Comment by imurchie** [comment]
The last is the correct way. Can you provide the logs for it?

**Comment by gauravgandhi15** [comment]
[debug] Pushing command to appium work queue: ["element:getText",{"elementId":"8"}]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getText","params":{"elementId":"8"}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: getText
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"\t\tEvents","status":0}
info: [debug] Responding to client with success: {"status":0,"value":"\t\tEvents","sessionId":"addf9d79-e6ba-48af-90f2-ade4b0a93d35"}
info: <-- GET /wd/hub/session/addf9d79-e6ba-48af-90f2-ade4b0a93d35/element/8/text 200 33.802 ms - 84 {"status":0,"value":"\t\tEvents","sessionId":"addf9d79-e6ba-48af-90f2-ade4b0a93d35"}
info: --> POST /wd/hub/session/addf9d79-e6ba-48af-90f2-ade4b0a93d35/touch/perform {"actions":[{"options":{"element":"8"},"action":"longPress"},{"options":{"element":"5"},"action":"moveTo"},{"options":{},"action":"release"}]}
info: [debug] Pushing command to appium work queue: ["element:getLocation",{"elementId":"8"}]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getLocation","params":{"elementId":"8"}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: getLocation
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"y":512,"x":16},"status":0}
info: [debug] Pushing command to appium work queue: ["element:getLocation",{"elementId":"5"}]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getLocation","params":{"elementId":"5"}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: getLocation
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"y":146,"x":16},"status":0}
info: [debug] Pushing command to appium work queue: ["element:drag",{"elementId":"8","destElId":"5","startX":16,"startY":512,"endX":16,"endY":146,"steps":40}]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:drag","params":{"elementId":"8","destElId":"5","startX":16,"startY":512,"endX":16,"endY":146,"steps":40}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: drag
info: [debug] [BOOTSTRAP] [debug] Dragging the element with id 8 to destination element with id 5 with steps: 40
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"addf9d79-e6ba-48af-90f2-ade4b0a93d35"}
info: <-- POST /wd/hub/session/addf9d79-e6ba-48af-90f2-ade4b0a93d35/touch/perform 200 1619.844 ms - 76 {"status":0,"value":true,"sessionId":"addf9d79-e6ba-48af-90f2-ade4b0a93d35"}
info: --> POST /wd/hub/session/addf9d79-e6ba-48af-90f2-ade4b0a93d35/back {}
info: [debug] Pushing command to appium work queue: ["pressBack"]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"pressBack","params":{}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: pressBack
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"addf9d79-e6ba-48af-90f2-ade4b0a93d35"}
info: <-- POST /wd/hub/session/addf9d79-e6ba-48af-90f2-ade4b0a93d35/back 200 1024.164 ms - 76 {"status":0,"value":true,"sessionId":"addf9d79-e6ba-48af-90f2-ade4b0a93d35"}
info: --> POST /wd/hub/session/addf9d79-e6ba-48af-90f2-ade4b0a93d35/back {}
info: [debug] Pushing command to appium work queue: ["pressBack"]
info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"pressBack","params":{}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: pressBack
info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
info: [debug] Responding to client with success: {"status":0,"value":true,"

**Comment by imurchie** [comment]
And that does nothing? This is invoking the drag and drop code in Android's ui automator. I'm not sure we can do much beyond that. 

If you could provide your app we could experiment. Beyond that...

**Comment by gauravgandhi15** [comment]
In the logs it says invoking "drag and drop" but actually in app it doesn't do anything

**Comment by imurchie** [comment]
Yes. This is, unfortunately, the command that Android gives us to use to automate drag and drop. For your application for whatever reason, this doesn't work. It's a matter of experimenting to find some solution that works.

**Comment by gauravgandhi15** [comment]
If i give you application, are you going to experiment it?

**Comment by imurchie** [comment]
I will. It would be nice to have different ways to do it, in case one doesn't work (like now!).

If it is private, you can email it to isaac at saucelabs dot com, and I won't share. Please also provide instructions on how to do the failing drag and drop.

**Comment by gauravgandhi15** [comment]
@imurchie Apk has been sent to your email

**Comment by imurchie** [comment]
I've replied to your email. In short, the issue is that the `longPress` is not long enough for lollipop. Just increase this by setting the `duration` parameter.

**Comment by DmitryKa** [comment]
Guys, help me please! I want to implement next behavior: tap&hold some element and move to another element, which apeears after holding of first element. How should I get it?
I tried

    Appium::TouchAction.new.press(element: my_elem).perform
    Appium::TouchAction.new.move_to(element: text(...)).perform

but between this actions pressing is disappeared. Also I tried

    Appium::TouchAction.new.press(element: my_elem).move_to(element: text(...)).perform

but in this case I have ```Selenium::WebDriver::Error::NoSuchElementError: No element found``` because element-destination for moving_to doesn't appeared yet. What the right way to implement it?

**Comment by imurchie** [comment]
Don't use `perform` for each. They should be chained.

```
Appium::TouchAction.new.press(element: my_elem).move_to(element: text(...)).perform
````

**Comment by DmitryKa** [comment]
@imurchie I tried this too, but error is raised. As I understood, checking of element existance (second element, ```text(...)```) executes immediately, but it appears on the screen only after pressing first element, so, I get ```Selenium::WebDriver::Error::NoSuchElementError: No element found```

**Comment by imurchie** [comment]
If the element isn't there when you begin, I'm not sure there is a way. You could drag to the location that it will be, I guess.

**Comment by DmitryKa** [comment]
It's hard, because element has not static position, it can appears in different places.

Is it impossible to check element existance not in the beginning of a chain, but only before appropriate action?

**Comment by DmitryKa** [comment]
I tried next way. Press element, waiting for neccessary elements appear and get their coordinates. Then I tried press again and move_to coordinates:

    Appium::TouchAction.new.press(element: ...).wait(3000).move_to(x: 253, y: 824).perform

But I got strange behavior: pressing, then releasing, then pressing in new coordinates. Is it bug in Appium working?

**Comment by Fanch-** [comment]
I have a similar problem, #4856 , so does that mean it's not possible to perform a drag&drop when the target element isn't present at the beginning?

**Comment by sebv** [comment]
@DmitryKa  please explain the strange behaviour.
Also why you don't release at the end?
@Fanch- it seems correct that you cannot press a hidden element, but you should be able to press by coordinate and move/release from there.

**Comment by Fanch-** [comment]
@sebv Thank you for the answer, I could do multiple longPress it's right, but it's not the real user Behavior. As far as I know, it's not possible to do press->move to Coordinate->waitForElement->move to anotherCoordinate -> waitForElement -> ElementIsPresent -> moveToElement -> Release -> Perform.

**Comment by sebv** [comment]
Maybe we should add some wait condition to the touch api (cc @jlipps ), otherwise you can try to do it using pure ios uiautomation and execute.

**Comment by DmitryKa** [comment]
@sebv  Now tested it in new Appium (previous was 1.3.6). Maybe something was fixed, maybe I didn't try this way, but now

    Appium::TouchAction.new.press(element: elem).wait(3000).move_to(element: text(58968.chr)).wait(3000).release.perform

seems work. As I remember, earlier problem was in releasing between `press` and `move_to`

**Comment by Fanch-** [comment]
Yes it didn"t work that way before, I'll try it again (Android).
And yes it could be great to have the possibility to waitFor Element during touchAction like press, longPress.

**Comment by sebv** [comment]
Feature: wait within touch action.

**Comment by Nitinvermaa1** [comment]
I am facing the similar problem. Below are the details.
Appium version: 1.4.0
Java_client: 3.1.0

Source code:

WebElement drag = driver.findElement(By.name("Gmail"));
		System.out.println(drag.isDisplayed());
		
		WebElement drop = driver.findElementByXPath("//android.widget.TextView[@text='']");
		Point location = drag.getLocation();
		Point location1=drop.getLocation();
		System.out.println(location.getX()+" "+location.getY());
		int x=location.getX();
		int y=location.getY();
		
		int x1=location1.getX();
		int y1=location1.getX()	;	
		Thread.sleep(2000);
		TouchAction action = new TouchAction(driver);
		//action.longPress(drag, x, y).moveTo(drop,x1,y1).release().perform();
		
		action.longPress(driver.findElement(By.name("Gmail"))).waitAction(5000);
		action.moveTo(drop,x1,y1).release();
		action.perform();

Error Log:

info: [debug] Pushing command to appium work queue: ["element:click",{"elementId":"8"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:click","params":{"elementId":"8"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: click
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/8/click 200 165.444 ms - 76 {"status":0,"value":true,"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element {"using":"name","value":"Installed Apps"}
> info: [debug] Waiting up to 0ms for condition
> info: [debug] Pushing command to appium work queue: ["find",{"strategy":"name","selector":"Installed Apps","context":"","multiple":false}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"find","params":{"strategy":"name","selector":"Installed Apps","context":"","multiple":false}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: find
> info: [debug] [BOOTSTRAP] [debug] Finding Installed Apps using NAME with the contextId:  multiple: false
> info: [debug] [BOOTSTRAP] [debug] Using: UiSelector[DESCRIPTION=Installed Apps, INSTANCE=0]
> info: [debug] [BOOTSTRAP] [debug] Using: UiSelector[TEXT=Installed Apps, INSTANCE=0]
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"ELEMENT":"9"},"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":{"ELEMENT":"9"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element 200 74.318 ms - 87 {"status":0,"value":{"ELEMENT":"9"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/9/click {"id":"9"}
> info: [debug] Pushing command to appium work queue: ["element:click",{"elementId":"9"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:click","params":{"elementId":"9"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: click
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/9/click 200 609.941 ms - 76 {"status":0,"value":true,"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element {"using":"name","value":"Gmail"}
> info: [debug] Waiting up to 0ms for condition
> info: [debug] Pushing command to appium work queue: ["find",{"strategy":"name","selector":"Gmail","context":"","multiple":false}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"find","params":{"strategy":"name","selector":"Gmail","context":"","multiple":false}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: find
> info: [debug] [BOOTSTRAP] [debug] Finding Gmail using NAME with the contextId:  multiple: false
> info: [debug] [BOOTSTRAP] [debug] Using: UiSelector[DESCRIPTION=Gmail, INSTANCE=0]
> info: [debug] [BOOTSTRAP] [debug] Using: UiSelector[TEXT=Gmail, INSTANCE=0]
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"ELEMENT":"10"},"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":{"ELEMENT":"10"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element 200 210.396 ms - 88 {"status":0,"value":{"ELEMENT":"10"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> GET /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/10/displayed {}
> info: [debug] Pushing command to appium work queue: ["element:getAttribute",{"elementId":"10","attribute":"displayed"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getAttribute","params":{"elementId":"10","attribute":"displayed"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getAttribute
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"true","status":0}
> info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- GET /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/10/displayed 200 567.258 ms - 76 {"status":0,"value":true,"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element {"using":"xpath","value":"//android.widget.TextView[@text='']"}
> info: [debug] Waiting up to 0ms for condition
> info: [debug] Pushing command to appium work queue: ["find",{"strategy":"xpath","selector":"//android.widget.TextView[@text='']","context":"","multiple":false}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"find","params":{"strategy":"xpath","selector":"//android.widget.TextView[@text='']","context":"","multiple":false}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: find
> info: [debug] [BOOTSTRAP] [debug] Finding //android.widget.TextView[@text=''] using XPATH with the contextId:  multiple: false
> info: [debug] [BOOTSTRAP] [debug] Using: UiSelector[CLASS=android.widget.TextView, INSTANCE=36]
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"ELEMENT":"11"},"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":{"ELEMENT":"11"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element 200 933.887 ms - 88 {"status":0,"value":{"ELEMENT":"11"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> GET /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/10/location {}
> warn: location will be removed in a future version of Appium, please use location_in_view
> info: [debug] Pushing command to appium work queue: ["element:getLocation",{"elementId":"10"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getLocation","params":{"elementId":"10"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getLocation
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"y":683,"x":199},"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":{"y":683,"x":199},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- GET /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/10/location 200 63.374 ms - 89 {"status":0,"value":{"y":683,"x":199},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> GET /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/11/location {}
> info: [debug] Pushing command to appium work queue: ["element:getLocation",{"elementId":"11"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getLocation","params":{"elementId":"11"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getLocation
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"y":362,"x":406},"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":{"y":362,"x":406},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- GET /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element/11/location 200 61.349 ms - 89 {"status":0,"value":{"y":362,"x":406},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element {"using":"name","value":"Gmail"}
> info: [debug] Waiting up to 0ms for condition
> info: [debug] Pushing command to appium work queue: ["find",{"strategy":"name","selector":"Gmail","context":"","multiple":false}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"find","params":{"strategy":"name","selector":"Gmail","context":"","multiple":false}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: find
> info: [debug] [BOOTSTRAP] [debug] Finding Gmail using NAME with the contextId:  multiple: false
> info: [debug] [BOOTSTRAP] [debug] Using: UiSelector[DESCRIPTION=Gmail, INSTANCE=0]
> info: [debug] [BOOTSTRAP] [debug] Using: UiSelector[TEXT=Gmail, INSTANCE=0]
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"ELEMENT":"12"},"status":0}
> info: [debug] Responding to client with success: {"status":0,"value":{"ELEMENT":"12"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/element 200 178.003 ms - 88 {"status":0,"value":{"ELEMENT":"12"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: --> POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/touch/perform {"actions":[{"action":"longPress","options":{"element":"12"}},{"action":"wait","options":{"ms":5000}},{"action":"moveTo","options":{"element":"11","x":406,"y":406}},{"action":"release","options":{}}]}
> info: [debug] Pushing command to appium work queue: ["element:getLocation",{"elementId":"11"}]
> info: [debug] Pushing command to appium work queue: ["element:getSize",{"elementId":"11"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getLocation","params":{"elementId":"11"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getLocation
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"y":362,"x":406},"status":0}
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getSize","params":{"elementId":"11"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getSize
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"height":22,"width":79},"status":0}
> info: [debug] Pushing command to appium work queue: ["element:getLocation",{"elementId":"12"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getLocation","params":{"elementId":"12"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getLocation
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"y":683,"x":199},"status":0}
> info: [debug] Pushing command to appium work queue: ["element:getSize",{"elementId":"12"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getSize","params":{"elementId":"12"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getSize
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"height":19,"width":80},"status":0}
> info: [debug] Pushing command to appium work queue: ["element:getLocation",{"elementId":"11"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getLocation","params":{"elementId":"11"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getLocation
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"y":362,"x":406},"status":0}
> info: [debug] Pushing command to appium work queue: ["element:getSize",{"elementId":"11"}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getSize","params":{"elementId":"11"}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getSize
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":{"height":22,"width":79},"status":0}
> info: [debug] Pushing command to appium work queue: ["element:touchLongClick",{"elementId":"12","x":239,"y":692.5,"duration":1000}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:touchLongClick","params":{"elementId":"12","x":239,"y":692.5,"duration":1000}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: touchLongClick
> info: [debug] [BOOTSTRAP] [debug] Performing TouchLongClick using element? true x: 239, y: 692, duration: 1000
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
> info: [debug] Pushing command to appium work queue: ["element:touchMove",{"elementId":"11","x":812,"y":768}]
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:touchMove","params":{"elementId":"11","x":812,"y":768}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: touchMove
> info: [debug] [BOOTSTRAP] [debug] Performing TouchMove using element? true x: 445, y: 373
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"Failed to execute touch event","status":13}
> info: [debug] Responding to client with error: {"status":13,"value":{"message":"An unknown server-side error occurred while processing the command.","origValue":"Failed to execute touch event"},"sessionId":"a82e1e2c-f21c-4e3d-92bc-7513423b8577"}
> info: <-- POST /wd/hub/session/a82e1e2c-f21c-4e3d-92bc-7513423b8577/touch/perform 500 6691.045 ms - 198 
> info: [debug] Didn't get a new command in 60 secs, shutting down...
> info: Shutting down appium session
> info: [debug] Pressing the HOME button
> info: [debug] executing cmd: C:\Users\nitin.verma\android-sdks\sdk\platform-tools\adb.exe -s QO4301C12715 shell "input keyevent 3"
> info: [debug] Stopping logcat capture
> info: [debug] Logcat terminated with code null, signal SIGTERM
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"shutdown"}
> info: [debug] [BOOTSTRAP] [debug] Got command of type SHUTDOWN
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"OK, shutting down","status":0}
> info: [debug] [UIAUTOMATOR STDOUT] [CDS]close[4724]
> info: [debug] [UIAUTOMATOR STDOUT] close [socket][/0.0.0.0:4724]
> info: [debug] [BOOTSTRAP] [debug] Closed client connection
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: current=1
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: id=UiAutomatorTestRunner
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: class=io.appium.android.bootstrap.Bootstrap
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: stream=.
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: numtests=1
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: test=testRunServer
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS_CODE: 0
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: stream=
> info: [debug] [UIAUTOMATOR STDOUT] Test results for WatcherResultPrinter=.
> info: [debug] [UIAUTOMATOR STDOUT] Time: 117.193
> info: [debug] [UIAUTOMATOR STDOUT] OK (1 test)
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS_CODE: -1
> info: [debug] Sent shutdown command, waiting for UiAutomator to stop...
> info: [debug] UiAutomator shut down normally
> info: [debug] Cleaning up android objects
> info: [debug] Cleaning up appium session
> info: [debug] We shut down because no new commands came in

**Comment by SrinivasanTarget** [comment]
Closing this because of inactivity.please raise in java-client if you hit the same.

**Comment by SrinivasanTarget** [comment]
@triager please close
